app/frontend/src/features/connections/friends/useRespondToFriendRequest.ts

```diff
@@ -0,0 +1,38 @@
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import { Error } from "grpc-web";
+import { useQueryClient, useMutation } from "react-query";
+import { SetMutationError } from ".";
+import { service } from "../../../service";
+
+interface RespondToFriendRequestVariables {
+  accept: boolean;
+  friendRequestId: number;
+  setMutationError: SetMutationError;
+}
+
+export default function useRespondToFriendRequest() {
+  const queryClient = useQueryClient();
+  const { mutate: respondToFriendRequest, isLoading, isSuccess } = useMutation<
+    Empty,
+    Error,
+    RespondToFriendRequestVariables
+  >(
+    ({ friendRequestId, accept }) =>
+      service.api.respondFriendRequest(friendRequestId, accept),
+    {
+      onMutate: async ({ setMutationError }) => {
+        setMutationError("");
+        await queryClient.cancelQueries("friendRequestsReceived");
+      },
+      onSuccess: () => {
```

Comment:
Do you want to optimistically update?
```suggestion
      onSuccess: () => {
        queryClient.removeQueries(["friendRequestsReceived", friendRequestId], true);
```

Reply:
could also do an optimistic add to the friends list if accepting.

Reply:
Hmm I saw react query had a section on this and thought about it, but didn't want to complicate the logic for this otherwise simple looking mutation.